src/Inventory/Asset/InventoryAsset.php

```diff
@@ -471,7 +471,23 @@ protected function handleInput(\stdClass $value, ?CommonDBTM $item = null): arra
                     $input[$key] = $this->known_links[$known_key];
                     $input['_raw' . $key] = $this->raw_links[$known_key];
                 } else {
-                    $input[$key] = $this->raw_links[$known_key];
+                    // If $item is new and the input key is locked, we do not want to set it using the raw value.
+                    // This is because locked fields are no longer processed or sanitized during the addition process.
+                    // For more details, see: https://github.com/glpi-project/glpi/pull/19426
+                    // Bypass for :
+                    // - states_id if default value is define from inventory conf
+                    // - entities_id default inventory conf is 0 'root entity'
+                    $config = \Config::getConfigurationValues('inventory');
```

Comment:
`$config` will be fetched from the DB for each value. It can result in performances issues.

If we could get the information from `$this->conf`, it would prevent any extra DB query. Currently the `$conf` property is private in many assets by the `public function checkConf(Conf $conf): bool` method. I do not know if there is a simple solution to make this property available in all assets without introducing a BC break.

Another solution is to move the `$config = \Config::getConfigurationValues('inventory');` line before the foreach loop.